### ISSUE TITLE
I am able to upload the code, but no output shown on my matrix

### ISSUE THREAD

**Issue by viralshah513** [JamieKitson/Ping-Pong-Matrix#1]
Hi Jamie, 

Thanks for the wonderful code and your videos on YouTube, I decided to change hardware of my clock to ESP32 as RTC module kept on failing with Nano. this is my first time with esp32 and I am more of an Hobbyist. 

I took your code, Pasted it to Arduino IDE, selected my ESP wroom 32 38 pin board. Compiled and got an error for <ESPAsyncWebServer.h> This specific library I was not able to installed and hence I installed this library ''https://github.com/dvarrel/ESPAsyncWebSrv"

Updated the code with my wifi and password and successfully uploaded to the board. Connected digital pin of led to pin 15
but led keeps on blinking randomly.

Apologies @JamieKitson for not mentioning the issue correctly previously. Its my first time on GIT as well. I have also drop a Instagram DM to you.

**my serial monitor output** 

ets Jul 29 2019 12:21:46

rst:0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0030,len:1344
load:0x40078000,len:13964
load:0x40080400,len:3600
entry 0x400805f0

it seems it is not getting connected to WIFI, I am not sure how to fix it 

**Video is here**
"[My setup video ](https://drive.google.com/file/d/1QVmML90R4GRgpxbmvZnNM7ssfrJO47ry/view?usp=sharing)"

Please help me in case I have missed some steps or if I am doing something wrong. 

Thanks in advance. 
once I get This working I also want to display message as per your youtube video( I will be elated if I can do that)

Best Regards, 
Viral Shah

**Comment by JamieKitson** [comment]
Hi Viral, 

I would take it one step at a time rather than jumping in and trying to use my code straight away. Start by checking that you can control the colours on the LEDS individually. Once you have done that you can then try to get messages working, etc. Note that my code won't actually work with your setup without modification because you have wired your LEDs differently to mine, you have wired yours the same as the indestructible project in a zigzag, up and down, whereas I wired mine all in the same direction and looped the wire around. This should make your code simpler as you won't need, for example, the invertAlternateColumns method.

Cheers, Jamie

**Comment by viralshah513** [comment]
Hey @JamieKitson Thanks for the quick response. 

I will rebuild as per your idea (which makes more sense to me now) can you let me know how many pixels you have used, was it 128 itself ?

And I am very knew to esp32 if you have any test codes for this project can you email me to 'viralshah513@gmail.com'

so my led pixel should look as below correct ? 
![IMG_8196](https://github.com/JamieKitson/Ping-Pong-Matrix/assets/9089282/51044806-a8e0-4f5b-8a4a-550f35421bb1)

Once this is done can I directly use your code ?

Super thankful for your Guidance

**Comment by JamieKitson** [comment]
Yes, that's how I have mine set out and yes, I used about 128 LEDs. The reason I set it out differently was because the clocks only use numbers always in the same place, so they can just use raw coordinates, however if you want to be able to use scrolling text libraries then I think it's probably easier having the LEDs sequential, although you might be able to use the remap function to work around that.

**Comment by viralshah513** [comment]
Alright, will do the modifications and will update it here.

Thanks  @JamieKitson

**Comment by viralshah513** [comment]
Hey @JamieKitson 

thanks I commented out the invertedchannel code and managed to run the code, Its working fine, Thanks ! 

i changed the postion of intial start point for hour as 0, minutes to 70 and colon to 62 and 64.

Can you let me know which part pf code gives italics numbers ?

same as your YouTube Video.

![IMG_8661](https://github.com/JamieKitson/Ping-Pong-Matrix/assets/9089282/0856a5ca-6aed-42f1-9451-d13f118cd436)

**Comment by JamieKitson** [comment]
I think I approached this project as a message board first, so I was using the FastLED mattrix message board to draw the numbers at this point and using the remap function to create the italic effect. The italic numbers don't fit nicely within the original dimensions of the board, so I use the numbers from the original project to draw the clock when there is no message.

**Comment by viralshah513** [comment]
@JamieKitson 
Thanks for all the help and guidance,
It is working perfectly for me will update pictures and video once it is completed.

can you guide me on how you approach the text and how you ended up using this approach.

**Comment by viralshah513** [comment]
@JamieKitson 
Greetings, I am not able to get desired output 
I have attached the images ,
I have connected leds left to right and sequence going from top to bottom , can you please help me ? 
also numbers are not visible correctly 

for me when i send Hi 
“i” comes inverted probable reason is you might have wired bottom to top, can you help me with the fix and my text scrolls on bottom 5 pixels how do i solve that

by any chance did you wire leds bottom to top ? 

![IMG_0549](https://github.com/JamieKitson/Ping-Pong-Matrix/assets/9089282/fbfe3ff9-77b6-4ebe-82bf-abd0336a620e)
![IMG_0548](https://github.com/JamieKitson/Ping-Pong-Matrix/assets/9089282/1a2118fa-757d-4bfd-93ad-705720dd4de9)

**Comment by viralshah513** [comment]
Hi Jamie,

i managed to get numbers correct, but i am unable to get text scrolling in center that is middle 5 rows, can you let me know what part of the code i am supposed to change ?
